src/create.cpp

```diff
@@ -40,25 +34,57 @@ unsigned Create::run()
         throw rest::Error(ret.error());
     }
 
-    std::string assetJson = m_request.body();
-
-    auto ret = AssetManager::createAsset(assetJson, user.login());
-    if (!ret) {
-        auditError(ret.error());
-        throw rest::errors::Internal(ret.error());
+    cxxtools::SerializationInfo si;
+    try {
+        std::stringstream          jsonIn(m_request.body());
+        cxxtools::JsonDeserializer deserializer(jsonIn);
+        deserializer.deserialize(si);
+    } catch (const std::exception& e) {
+        auditError(e.what());
+        throw rest::errors::Internal(e.what());
     }
 
-    auto createdAsset = AssetManager::getItem(*ret);
-
-    if (!createdAsset) {
-        auditError(createdAsset.error());
-        throw rest::errors::Internal(createdAsset.error());        
-    } 
-
-    m_reply << "{ " << utils::json::jsonify("id", createdAsset->name) << " }\n\n";
+    bool             someOk = false;
```

Comment:
Existence of someOk is under of question. You could just check if something is in list.